classes/patterns/class-base-pattern.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Base pattern class.
+ *
+ * @package P4GBKS
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * Class Base_Pattern
+ *
+ * @package P4GBKS\Patterns
+ */
+abstract class Base_Pattern {
+
+	public const NAMESPACE = 'planet4-patterns';
```

Comment:
It's not needed to define this namespace separately, as it will only be used inside the name. So it's a lot easier to just put the namespace in the name.

Reply:
Can you remove this namespace and include it directly in the block name?

Also, similar to with block name prefixes, we could go with a short prefix `p4/`.